pyproject.toml

```diff
@@ -0,0 +1,144 @@
+[build-system]
+requires = ["hatchling"]
+build-backend = "hatchling.build"
+
+[project]
+name = "datasketch"
+dynamic = ["version"]
+description = "Probabilistic data structures for processing and searching very large datasets"
+readme = "README.rst"
+requires-python = ">=3.9"
+license = { text = "MIT" }
+authors = [{ name = "ekzhu", email = "ekzhu@cs.toronto.edu" }]
+keywords = ["database", "datamining"]
+classifiers = [
+  "Development Status :: 5 - Production/Stable",
+  "Intended Audience :: Developers",
+  "Topic :: Database",
+  "Topic :: Scientific/Engineering :: Information Analysis",
+  "License :: OSI Approved :: MIT License",
+  "Programming Language :: Python :: 3.9",
+  "Programming Language :: Python :: 3.10",
+  "Programming Language :: Python :: 3.11",
+  "Programming Language :: Python :: 3.12",
+]
+dependencies = ["numpy>=1.11", "scipy>=1.0.0"]
+
+[project.optional-dependencies]
+cassandra = ["cassandra-driver>=3.20"]
+redis = ["redis>=2.10.0"]
+bloom = ["pybloomfilter3>=0.7.2"]
+benchmark = [
+  "pyhash>=0.9.3",
+  "matplotlib>=3.1.2",
+  "scikit-learn>=0.21.3",
+  "scipy>=1.3.3",
+  "pandas>=0.25.3",
+  "SetSimilaritySearch>=0.1.7",
+  "pyfarmhash>=0.2.2",
+  "nltk>=3.4.5",
+]
+test = [
+  "cassandra-driver>=3.20",
+  "redis>=2.10.0",
+  "mock>=2.0.0",
+  "mockredispy",
+  "coverage",
+  "pymongo>=3.9.0",
+  "nose>=1.3.7",
+  "nose-exclude>=0.5.0",
+  "pytest",
+  "pytest-rerunfailures",
+]
+experimental_aio = ["aiounittest", "motor>3.6.0"]
+
+[project.urls]
+Homepage = "https://ekzhu.github.io/datasketch"
+"Bug Tracker" = "https://github.com/ekzhu/datasketch/issues"
+Documentation = "https://ekzhu.github.io/datasketch"
+Source = "https://github.com/ekzhu/datasketch"
+
+[tool.codespell]
+# Todo: enable also python files in a next step
+# skip = '*.py'
+quiet-level = 3
+# comma separated list of words; waiting for:
+# https://github.com/codespell-project/codespell/issues/2839#issuecomment-1731601603
+# also adding links until they ignored by its: nature
+# https://github.com/codespell-project/codespell/issues/2243#issuecomment-1732019960
+ignore-words-list = "cancelation"
+
+[tool.docformatter]
+recursive = true
+# this need to be shorter as some docstings are r"""...
+wrap-summaries = 119
+wrap-descriptions = 120
+blank = true
+
+[tool.ruff]
+line-length = 120
+target-version = "py39"
+exclude = [
+  ".eggs",
+  ".git",
+  ".mypy_cache",
+  ".ruff_cache",
+  "__pycache__",
+  "__pypackages__",
+  "_build",
+  "benchmark",
+  "build",
+  "dist",
+  "docs",
+  "examples",
+  "travis",
+  "datasketch/hyperloglog_const.py",
+]
+
+[tool.ruff.lint]
+select = [
+  "E",
+  "W",      # see: https://pypi.org/project/pycodestyle
+  "F",      # see: https://pypi.org/project/pyflakes
+  "N",      # see: https://pypi.org/project/pep8-naming
+  "I",      # implementation for isort
+  "UP",     # implementation for pyupgrade
+  "RUF100", # implementation for yesqa
+]
+ignore = [
```

Comment:
For lint ignore settings I suggest we copy from https://github.com/microsoft/agent-framework/blob/1aaf37dab8ee4534ae2000082a08ba112173cc4c/python/pyproject.toml#L116

Reply:
Sure @ekzhu, done. I’ve also added a few TODOs that should be addressed in a separate PR so they don’t block this one.